docs/getting_started/upgrading.md

```diff
@@ -53,6 +53,196 @@ It skips any rows with an "invalid" `resc_id` value. A `resc_id` is considered i
 
 The script has a `--dry-run` option and can safely be interrupted and run again (it is only querying the database for values to update, and then updating them).
 
+## Migrating from Static PEPs to Dynamic PEPs
+
+The dynamic policy enforcement points (PEPs) represent the most reliable way to interact with the iRODS rule engine plugin framework (REPF).  Migrating from the legacy static policy enforcement points (PEPs) is recommended.
+
+Migrating involves understanding the mapping of the legacy session variables to the dynamic context-based variables.  Please see the examples below for how the REPF provides the full operating context of each dynamic PEP and how they can be used in the rule and policy logic.
+
+### acPostProcForPut()
+
+Using the static PEP
+```
+acPostProcForPut(){
+    writeLine("serverLog",$rescName)
+    writeLine("serverLog",$objPath)
+    writeLine("serverLog",$userNameClient)
+    writeLine("serverLog",$rodsZoneClient)
+    writeLine("serverLog",$dataSize)
+}
+```
+produces
+```
+Apr 19 14:56:44 pid:22819 NOTICE: writeLine: inString = anotherResc
+Apr 19 14:56:44 pid:22819 NOTICE: writeLine: inString = /tempZone/home/rods/testfile
+Apr 19 14:56:44 pid:22819 NOTICE: writeLine: inString = rods
+Apr 19 14:56:44 pid:22819 NOTICE: writeLine: inString = tempZone
+Apr 19 14:56:44 pid:22819 NOTICE: writeLine: inString = 12
+```
+
+### pep_api_data_obj_put_post()
+
+Moving to the dynamic PEP that fires after a data object is put via the iRODS API
+```
+pep_api_data_obj_put_post(*INSTANCE_NAME, *COMM, *DATAOBJINP, *BUFFER, *PORTAL_OPR_OUT){
+    writeLine("serverLog",*DATAOBJINP)
```

Comment:
Should we also demonstrate the output from *COMM to show user related information?

Reply:
There is a *COMM example farther down...

https://github.com/irods/irods_docs/pull/46/files#diff-b0492b86f9fe84c51f7c359012ff8f25R232